app/src/main/java/fr/free/nrw/commons/contributions/RxJava2Tasks.java

```diff
@@ -0,0 +1,30 @@
+package fr.free.nrw.commons.contributions;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Locale;
+
+import fr.free.nrw.commons.PageTitle;
+import io.reactivex.Single;
+
+class RxJava2Tasks {
+
+    private static final String UPLOAD_COUNT_URL_TEMPLATE =
+            "https://tools.wmflabs.org/urbanecmbot/uploadsbyuser/uploadsbyuser.py?user=%s";
+
+    static Single<Integer> getUploadCount(String userName) {
+        return Single.fromCallable(() -> {
+            URL url = new URL(String.format(Locale.ENGLISH, UPLOAD_COUNT_URL_TEMPLATE,
+                    new PageTitle(userName).getText()));
+            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+            BufferedReader bufferedReader = new BufferedReader(new
+                    InputStreamReader(urlConnection.getInputStream()));
+            String uploadCount = bufferedReader.readLine();
```

Comment:
_Leaking_ the exception is probably a more loaded phrase and more combative than I really wanted, sorry.

I think it boils down to the `NULL` being _expected_ behaviour rather than being _exceptional_ - the `NULL` has meaning (there was no content) and is documented normal behaviour of the `Reader`.  In that sense I am splitting hairs and getting philosophical over a difference that in the grand scheme isnt a big deal.

Its the difference between using `Single` and `Observable` - we can use the richer subscription of `Observable` to communicate the meaning of the `NULL` value.  If we have no content then it feels like the code should simply skip the `onNext` and call `onComplete` as opposed to allowing it to blow up the conversion to integer that follows; the log from the parsing error doesnt immediately tell me that I received no content, whereas an explicit NULL check plus log would.  

The thought about "leaking" comes from a sense of whose responsibility it is to know about the lack of content and separation of concerns - the GUI is being handed an NPE but doesnt have context to know it.  It feels to me that its the responsibility of the code close to the URL connection to handle content issues and communicate in a meaningful way back to the GUI.

Sorry about the wall of text, hope this helps.